File: frontend/src/components/Question.js
import styled from 'styled-components/macro'
import Answer from './Answer'

export default function Question({ question }) {
  return (
    <QuestionContainer>
      <hr />
      <QuestionText>{question.questionText}</QuestionText>
      <AnswerContainer>
        {question.answers.map(answer => (
          <Answer
            answer={answer}
            questionID={question.questionID}
            key={answer.answerID}
          />
        ))}
      </AnswerContainer>
      <CheckButton>Check Answer</CheckButton>
      <hr />
    </QuestionContainer>
  )
}
const QuestionContainer = styled.section``
const QuestionText = styled.p``
const AnswerContainer = styled.section`
  display: flex;
  flex-wrap: wrap;
`
const CheckButton = styled.button`
  border-radius: 5px;
`

File: backend/src/main/java/de/neuefische/backend/model/Question.java
package de.neuefische.backend.model;

import lombok.Data;

import java.util.List;

@Data
public class Question {
    private String questionID;
    private String questionText;
    private List<Answer> answers;

    public Question(String questionID, String questionText, List<Answer> answers) {
        this.questionID = questionID;
        this.questionText = questionText;
        this.answers = answers;
    }

    public Question(String questionText, List<Answer> answers) {
        this.questionText = questionText;
        this.answers = answers;
    }

    public Question() {
    }
}
